src/vm.rs

```diff
@@ -576,10 +575,16 @@ pub trait Vm {
 			write(&mut (*boot_info).hcmask, mask.octets());
 		}
 
-		// TODO: should be a random start address
-		let start_address: u64 = 0x400000;
-		self.set_entry_point(start_address + elf.entry);
-		debug!("ELF entry point at 0x{:x}", start_address + elf.entry);
+		let (start_address, elf_entry) = if is_dyn {
+			// TODO: should be a random start address, if we have a relocatable executable
+			(0x400000u64, 0x400000u64 + elf.entry)
+		} else {
+			// default location of a non-relocatable binary
+			(0x800000u64, elf.entry)
+		};
+
+		self.set_entry_point(elf_entry);
+		debug!("ELF entry point at 0x{:x}", elf_entry);
```

Comment:
If you have to change anything regarding my previous comment, feel free to also add a printout that shows the value of is_dyn.

Reply:
Good idea.